pydra/engine/core.py

```diff
@@ -439,7 +439,8 @@ def _run(self, rerun=False, **kwargs):
             if not (rerun or self.task_rerun):
                 result = self.result()
                 if result is not None:
-                    return result
+                    if not result.errored:
+                        return result
```

Comment:
perhaps change to:

```
if result is not None and not result.errored:
    return result
```

both here and in the next patch.

Reply:
That is a good fix. It does the same thing and is more concise.